pytorch_lightning/strategies/ddp.py

```diff
@@ -232,23 +236,18 @@ def _register_ddp_hooks(self) -> None:
                 import torch.distributed.algorithms.ddp_comm_hooks.post_localSGD_hook as post_localSGD
 
                 if isinstance(self._ddp_comm_state, post_localSGD.PostLocalSGDState):
-                    self._reinit_optimizers_with_post_localSGD(self._ddp_comm_state.start_localSGD_iter)
+                    self._enable_model_averaging()
```

Comment:
I see, I guess this tightly couples localSGD state + DDP comm hook to model averaging component, this should be fine at the moment, although @wayi1 is also working on hierarchical SGD which might require additional design thought to integrate into lightning.

Reply:
sounds good!

Reply:
Thanks for bringing this up @rohan-varma. This should be compatible with hierarchical SGD, which will still use localSGD state + DDP comm hook.